h2o-core/src/main/java/water/util/Checksum.java

```diff
@@ -0,0 +1,107 @@
+package water.util;
+
+import water.Checksumable;
+import water.H2O;
+import water.Weaver;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Set;
+
+public final class Checksum {
+  
+  private Checksum() {}
+
+  public static <T> long checksum(final T obj) {
+    return checksum(obj, null);
+  }
+  
+  public static <T> long checksum(final T obj, final Set<String> ignoredFields) {
+    return checksum(obj, ignoredFields, 0x600DL);
+  }
+  
+  /**
+   * Compute a checksum based on all non-transient non-static ice-able assignable fields (incl. inherited ones).
+   * Sort the fields first, since reflection gives us the fields in random order, and we don't want the checksum to be affected by the field order.
+   * 
+   * NOTE: if a field is added to a class the checksum will differ even when all the previous parameters have the same value.  If
+   *       a client wants backward compatibility they will need to compare values explicitly.
+   *
+   * The method is motivated by standard hash implementation `hash = hash * P + value` but we use high prime numbers in random order.
+   * 
+   * @param ignoredFields A {@link Set} of fields to ignore. Can be empty or null.
+   * @return checksum A 64-bit long representing the checksum of the object
+   */
+  public static <T> long checksum(final T obj, final Set<String> ignoredFields, final long initVal) {
```

Comment:
logic extracted for `Model.Parameters` so that it can also be used in pipeline `DataTransformer`s